File: core/src/main/java/org/dockbox/darwin/core/events/AbstractTargetCancellableEvent.java
package org.dockbox.darwin.core.events;

import org.dockbox.darwin.core.objects.events.Cancellable;
import org.dockbox.darwin.core.objects.events.Targetable;
import org.dockbox.darwin.core.objects.targets.Target;
import org.jetbrains.annotations.NotNull;

public class AbstractTargetCancellableEvent implements Cancellable, Targetable {

    private boolean isCancelled;
    private Target target;

    public AbstractTargetCancellableEvent(Target target) {
        this.target = target;
    }

    @Override
    public boolean isCancelled() {
        return this.isCancelled;
    }

    @Override
    public void setCancelled(boolean cancelled) {
        this.isCancelled = cancelled;
    }

    @NotNull
    @Override
    public Target getTarget() {
        return this.target;
    }

    @Override
    public void setTarget(@NotNull Target target) {
        this.target = target;
    }
}

File: core/src/main/kotlin/org/dockbox/darwin/core/events/AbstractTargetCancellableEvent.kt
package org.dockbox.darwin.core.events

import org.dockbox.darwin.core.objects.events.Cancellable
import org.dockbox.darwin.core.objects.events.Targetable
import org.dockbox.darwin.core.objects.targets.Target

abstract class AbstractTargetCancellableEvent(private var target: Target) : Cancellable, Targetable {
    private var isCancelled = false
    override fun isCancelled(): Boolean {
        return isCancelled
    }

    override fun setCancelled(cancelled: Boolean) {
        isCancelled = cancelled
    }

    override fun getTarget(): Target {
        return target
    }

    override fun setTarget(target: Target) {
        this.target = target
    }

}
